pkg/operator/resourcesynccontroller/resourcesync_controller.go

```diff
@@ -0,0 +1,221 @@
+}
+
+var _ ResourceSyncer = &ResourceSyncController{}
+
+// NewResourceSyncController creates a new backing resource controller.
+func NewResourceSyncController(
+	operatorConfigClient common.OperatorClient,
+	// kubeInformersForNamespaces is a map of namespaces to their informers
+	kubeInformersForNamespaces map[string]informers.SharedInformerFactory,
+	kubeClient kubernetes.Interface,
+	eventRecorder events.Recorder,
+) *ResourceSyncController {
+	c := &ResourceSyncController{
+		operatorConfigClient: operatorConfigClient,
+		eventRecorder:        eventRecorder,
+
+		configMapSyncRules: map[ResourceLocation]ResourceLocation{},
+		secretSyncRules:    map[ResourceLocation]ResourceLocation{},
+		knownNamespaces:    sets.StringKeySet(kubeInformersForNamespaces),
+
+		queue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ResourceSyncController"),
+		kubeClient: kubeClient,
+	}
+
+	for _, informers := range kubeInformersForNamespaces {
+		informers.Core().V1().ConfigMaps().Informer().AddEventHandler(c.eventHandler())
+		informers.Core().V1().Secrets().Informer().AddEventHandler(c.eventHandler())
+		c.preRunCachesSynced = append(c.preRunCachesSynced, informers.Core().V1().ConfigMaps().Informer().HasSynced)
+		c.preRunCachesSynced = append(c.preRunCachesSynced, informers.Core().V1().Secrets().Informer().HasSynced)
+	}
+
+	// we watch this just in case someone messes with our status
+	operatorConfigClient.Informer().AddEventHandler(c.eventHandler())
+
+	return c
+}
+
+func (c *ResourceSyncController) SetConfigMapSyncRule(destination, source ResourceLocation) error {
+	if !c.knownNamespaces.Has(destination.Namespace) {
+		return fmt.Errorf("not watching namespace %q", destination.Namespace)
+	}
+	if !c.knownNamespaces.Has(source.Namespace) {
+		return fmt.Errorf("not watching namespace %q", source.Namespace)
+	}
+
+	c.syncRuleLock.Lock()
+	defer c.syncRuleLock.Unlock()
+	c.configMapSyncRules[destination] = source
+	return nil
+}
+
+func (c *ResourceSyncController) SetSecretSyncRule(destination, source ResourceLocation) error {
+	if !c.knownNamespaces.Has(destination.Namespace) {
+		return fmt.Errorf("not watching namespace %q", destination.Namespace)
+	}
+	if !c.knownNamespaces.Has(source.Namespace) {
+		return fmt.Errorf("not watching namespace %q", source.Namespace)
+	}
+
+	c.syncRuleLock.Lock()
+	defer c.syncRuleLock.Unlock()
+	c.secretSyncRules[destination] = source
+	return nil
+}
+
+func (c *ResourceSyncController) sync() error {
+	operatorSpec, _, _, err := c.operatorConfigClient.Get()
+	if err != nil {
+		return err
+	}
+
+	switch operatorSpec.ManagementState {
+	case operatorv1.Unmanaged:
+		return nil
+	case operatorv1.Removed:
+		// TODO: Should we try to actively remove the resources created by this controller here?
+		return nil
+	}
+
+	c.syncRuleLock.RLock()
+	defer c.syncRuleLock.RUnlock()
+
+	errors := []error{}
+
+	for destination, source := range c.configMapSyncRules {
```

Comment:
`from, to` would make it easier to read.

Reply:
> `from, to` would make it easier to read.

golang `copy` method uses dest,src

Reply:
also fine. even d, s.